brave-instrumentation/aws-java-sdk-core/src/main/java/brave/instrumentation/aws/HttpAdapter.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2016-2018 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.instrumentation.aws;
+
+import brave.http.HttpClientAdapter;
+import com.amazonaws.Request;
+import com.amazonaws.Response;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import zipkin2.Endpoint;
+
+public class HttpAdapter extends HttpClientAdapter<Request<?>, Response<?>> {
+  @Override public boolean parseServerAddress(Request<?> request, Endpoint.Builder builder) {
+    // Does this belong here?
+    builder.serviceName(request.getServiceName());
+
+    InetAddress inetAddress = null;
+    try {
+      inetAddress = InetAddress.getByName(request.getEndpoint().getHost());
+    } catch (UnknownHostException e) {
+      return false;
+    }
+    return builder.parseIp(inetAddress);
+  }
+
+  @Override public String method(Request<?> request) {
```

Comment:
might override path, too, if it is cheap

Reply:
The path is always empty or `/` since all their work is done by the RPC nature of the payload. Wondering if method should be the RPC method, and path should be the service name?